providers/os/connection/device/linux/lun.go

```diff
@@ -22,6 +25,49 @@ type scsiDeviceInfo struct {
 type scsiDevices = []scsiDeviceInfo
 
 func (c *LinuxDeviceManager) listScsiDevices() ([]scsiDeviceInfo, error) {
+	devices, err := c.listScsiDevicesFromSys()
+	if err == nil {
+		return devices, nil
+	}
+
+	log.Warn().Err(err).Msg("failed to list scsi devices from sys, trying lsscsi")
+	return c.listScsiDevicesFromCLI()
+}
+
+func (c *LinuxDeviceManager) listScsiDevicesFromSys() ([]scsiDeviceInfo, error) {
+	blocks, err := os.ReadDir("/sys/block/")
+	if err != nil {
+		return nil, err
+	}
+	scsiDevices := []scsiDeviceInfo{}
+	for _, block := range blocks {
+		if block.Type() != os.ModeSymlink {
+			continue
+		}
+
+		entry, err := os.Readlink(path.Join("/sys/block/", block.Name()))
+		if err != nil {
+			log.Warn().Err(err).Str("block", block.Name()).Msg("failed to readlink")
+			continue
+		}
+		entry, err = filepath.Abs(path.Join("/sys/block/", entry))
+		if err != nil {
+			log.Warn().Err(err).Str("block", block.Name()).Msg("failed to get absolute path")
```

Comment:
do we need these logs to be warnings?

Reply:
Something has to be very broken for these to fail and produce a log, it'd be very helpful to have that log.